x/aswap/model.go

```diff
@@ -89,8 +95,24 @@ func NewBucket() Bucket {
 		WithIndex("preimage_hash", idxPrehash, false)
 
 	return Bucket{
-		IDGenBucket: orm.WithSeqIDGenerator(bucket, SequenceName),
+		Bucket: bucket,
+		idSeq:  bucket.Sequence("id"),
+	}
+}
+
+// Create adds given swap instance to the store and returns the ID of the newly
+// inserted entity.
+func (b *Bucket) Create(db weave.KVStore, swap *Swap) (orm.Object, error) {
+	key, err := b.idSeq.NextVal(db)
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot acquire ID")
+	}
+	swap.Address = SwapAddr(key, swap)
```

Comment:
I actually did utilise this approach when working on the extension originally and I do remember it receiving some critique due to the fact that the logic that defines what is in the object is spread between several places, therefore a bit "magical". 

It is the easiest way to implement this here, however. 

I would suggest to add an extra method here that would allow to peek too, in order to avoid any inconsistencies throughout our extension and confusion while using them. And then set this Address in the handler. 

One reason for that would be the following: imagine someone new comes and wants to write an extension. And then it's a dice roll depending on which extension that person follows as they utilise different approaches. How does one know which one is right?

Again, I'm not really criticising, as this is would be the first thing I'd do to make it work for `aswap` and I have actually already done it before, although none of that made it to master.